common/djangoapps/student/models.py

```diff
@@ -884,7 +916,9 @@ def is_course_full(self, course):
         """
         is_course_full = False
         if course.max_student_enrollments_allowed is not None:
-            is_course_full = self.num_enrolled_in(course.id) >= course.max_student_enrollments_allowed
+            is_course_full = (
+               self.num_enrolled_in_exclude_admins(course.id) >= course.max_student_enrollments_allowed
+            )
```

Comment:
Sorry if my question in standup was confusing. This is exactly what I had in mind -- the is_course_full() check excludes admins.